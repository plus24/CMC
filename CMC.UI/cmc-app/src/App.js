import "./App.css";
import Products from "./components/Products/Products";
import Navbar from "./components/NavBar";
import { useState } from "react";
import BasketContextProvider from "./contexts/BasketContext";
import CheckOut from "./components/CheckOut";
import CurrencyContextProvider from "./contexts/CurrencyContext";
import { AppBar, Toolbar, Typography } from "@mui/material";
import { Box } from "@mui/system";

function App() {
  const [isCheckOut, setCheckOut] = useState(false);
  const toggleCheckOut = () => {
    setCheckOut(!isCheckOut);
  };

  return (
    <CurrencyContextProvider>
      <BasketContextProvider className="App">
        <Box sx={{ flexGrow: 1 }}>
          <AppBar position="static">
          <Toolbar>
          <Typography variant="h6" component="div" sx={{ flexGrow: 1 }}>
            CMC Market
          </Typography>
          <Navbar
            toggleCheckOut={toggleCheckOut}
            isCheckOut={isCheckOut}
          ></Navbar>
        </Toolbar>
          </AppBar>
        </Box>
        <section className="body">
          {isCheckOut ? (
            <CheckOut setCheckOut={setCheckOut} ></CheckOut>
          ) : (
            <Products></Products>
          )}
        </section>
      </BasketContextProvider>
    </CurrencyContextProvider>
  );
}

export default App;
