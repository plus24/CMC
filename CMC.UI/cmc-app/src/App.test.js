import React from 'react';
import renderer from 'react-test-renderer';
import App from './App';

test('renders App without crashing', () => {
  const component = renderer.create(
      <App currency="AUD"></App>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})