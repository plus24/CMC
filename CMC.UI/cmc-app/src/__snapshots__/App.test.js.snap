// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App without crashing 1`] = `
Array [
  <div
    className="MuiBox-root css-i9gxme"
  >
    <header
      className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
    >
      <div
        className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1t29gy6-MuiToolbar-root"
      >
        <div
          className="MuiTypography-root MuiTypography-h6 css-1juivf6-MuiTypography-root"
        >
          CMC Market
        </div>
        <div
          className="rightNav"
        >
          <div
            className="basket"
          >
            <div
              className="basket"
            >
              <span
                className="MuiBadge-root css-1c32n2y-MuiBadge-root"
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-bag-shopping fa-2x "
                  data-icon="bag-shopping"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  style={Object {}}
                  viewBox="0 0 448 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M112 112C112 50.14 162.1 0 224 0C285.9 0 336 50.14 336 112V160H400C426.5 160 448 181.5 448 208V416C448 469 405 512 352 512H96C42.98 512 0 469 0 416V208C0 181.5 21.49 160 48 160H112V112zM160 160H288V112C288 76.65 259.3 48 224 48C188.7 48 160 76.65 160 112V160zM136 256C149.3 256 160 245.3 160 232C160 218.7 149.3 208 136 208C122.7 208 112 218.7 112 232C112 245.3 122.7 256 136 256zM312 208C298.7 208 288 218.7 288 232C288 245.3 298.7 256 312 256C325.3 256 336 245.3 336 232C336 218.7 325.3 208 312 208z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
                <span
                  className="MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular MuiBadge-colorWarning MuiBadge-badge MuiBadge-standard MuiBadge-anchorOriginTopRight MuiBadge-invisible css-17v1rux-MuiBadge-badge"
                />
              </span>
            </div>
          </div>
          <div
            className="currencies"
          >
            <div
              className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary currencies css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
              onClick={[Function]}
            >
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby="demo-simple-select-label demo-simple-select"
                className="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                id="demo-simple-select"
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                role="button"
                tabIndex={0}
              >
                AUD
              </div>
              <input
                aria-hidden={true}
                className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                onAnimationStart={[Function]}
                onChange={[Function]}
                tabIndex={-1}
                value="AUD"
              />
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-173wfxe"
                >
                  <span>
                    Currency
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
    </header>
  </div>,
  <section
    className="body"
  >
    <div>
      <div>
        <span
          className="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18lrjg1-MuiCircularProgress-root"
          role="progressbar"
          style={
            Object {
              "height": 40,
              "width": 40,
            }
          }
        >
          <svg
            className="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
            viewBox="22 22 44 44"
          >
            <circle
              className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
              cx={44}
              cy={44}
              fill="none"
              r={20.2}
              strokeWidth={3.6}
              style={Object {}}
            />
          </svg>
        </span>
      </div>
    </div>
  </section>,
]
`;
